### ISSUE TITLE
Issues for using getvar() with files contain multiple timeidx or without 'XTIME'

### ISSUE THREAD

**Issue by yingkaisha** [NCAR/wrf-python#11]
Hi all,
I got a problem when using wrf.getvar to read variables from netCDF4 objects for both single and multiple time cases.

===========================================================
```
D02 = nc.Dataset('wrfout_d02_2016-08-19_06:00:00')
slp_02 = wrf.getvar(D02, 'slp’)
# The file is readable via netCDF4
```
==================================================================
```
KeyError                                  Traceback (most recent call last)
<ipython-input-3-eb8ee2965f9c> in <module>()
---> 14 slp_02 = wrf.getvar(D02, 'slp')
......
/Users/yingkai/anaconda2/lib/python2.7/site-packages/wrf/util.pyc in _build_data_array(wrfnc, varname, timeidx, is_moving_domain, is_multifile, _key)
   1237 
   1238                 if time_coord_vals is None:
-> 1239                     time_coord_vals = wrfnc.variables[time_coord][:]
   1240 
   1241                     if not is_multifile:

KeyError: u'XTIME'
```
=======================================================
```
timeidx = range(31)
ua = wrf.getvar(nc_obj, 'ua', timeidx=timeidx)
```
=======================================================
```
<ipython-input-2-324fca9a34ef> in wrf_reader_2d(filename, wind_plev)
---> 65     ua = wrf.getvar(nc_obj, 'ua', timeidx=timeidx)
/anaconda2/lib/python2.7/site-packages/wrf/routines.pyc in getvar(wrfin, varname, timeidx, method, squeeze, cache, meta, **kwargs)
    283 
    284     return _FUNC_MAP[actual_var](wrfin, timeidx, method, squeeze, cache, 
--> 285                                  meta, _key, **kwargs)
    286 
/anaconda2/lib/python2.7/site-packages/wrf/metadecorators.pyc in func_wrapper(wrapped, instance, args, kwargs)
    260         copy_var = extract_vars(wrfin, timeidx, _copy_varname, 
    261                                 method, squeeze, cache,
--> 262                                 meta=True, _key=_key)[_copy_varname]
    263 
    264         # Make a copy so we don't modify a user supplied cache
/anaconda2/lib/python2.7/site-packages/wrf/util.pyc in extract_vars(wrfin, timeidx, varnames, method, squeeze, cache, meta, _key)
   2172     return {var:_extract_var(wrfin, var, timeidx, None,
   2173                              method, squeeze, cache, meta, _key)
-> 2174             for var in varlist}
   2175 
   2176 
/anaconda2/lib/python2.7/site-packages/wrf/util.pyc in <dictcomp>((var,))
   2172     return {var:_extract_var(wrfin, var, timeidx, None,
   2173                              method, squeeze, cache, meta, _key)
-> 2174             for var in varlist}
   2175 
   2176 
/anaconda2/lib/python2.7/site-packages/wrf/util.pyc in _extract_var(wrfin, varname, timeidx, is_moving, method, squeeze, cache, meta, _key)
   2095                 is_moving = is_moving_domain(wrfin, varname, _key=_key)
   2096             result = _build_data_array(wrfin, varname, timeidx, is_moving,
-> 2097                                        multifile, _key)
   2098         else:
   2099             if not multitime:
/anaconda2/lib/python2.7/site-packages/wrf/util.pyc in _build_data_array(wrfnc, varname, timeidx, is_moving_domain, is_multifile, _key)
   1270 
   1271         proj_params = get_proj_params(wrfnc)
-> 1272         proj = getproj(**proj_params)
   1273         attrs["projection"] = proj
   1274 
/anaconda2/lib/python2.7/site-packages/wrf/projection.pyc in getproj(**proj_params)
   1081         return LambertConformal(**proj_params)
   1082     elif proj_type == ProjectionTypes.POLAR_STEREOGRAPHIC:
-> 1083         return PolarStereographic(**proj_params)
   1084     elif proj_type == ProjectionTypes.MERCATOR:
   1085         return Mercator(**proj_params)
/anaconda2/lib/python2.7/site-packages/wrf/projection.pyc in __init__(self, **proj_params)
    676 
    677         """
--> 678         super(PolarStereographic, self).__init__(bottom_left, 
    679                         top_right, lats, lons, **proj_params)
    680         self._hemi = -90. if self.truelat1 < 0 else 90.

NameError: global name 'bottom_left' is not defined
```
==========================================================
Actually I was working ok last week, but I did a 
conda update --all
for installing gdal and other packages. not sure if that raises the problem.
 
Thanks

**Comment by bladwig1** [comment]
Hi yingkaisha,

I'm working on a hotfix for the projection 'bottom_left' error, but need some more information for the other problems.  Are you able to send us wrfout_d02_2016-08-19_06:00:00 following the instructions at http://wrf-python.readthedocs.io/en/latest/support.html ?  If that's not an option, can you show the result from doing an 'ncdump -h wrfout_d02_2016-08-19_06:00:00' .  I'm not sure how you're getting a KeyError for 'XTIME', since 'XTIME' has to be found in the variable list before it tries to use it.  I'd need to look at the data to see what's going on.

Also, this code:

> timeidx = range(31)
   ua = wrf.getvar(nc_obj, 'ua', timeidx=timeidx)

isn't valid, since timeidx can only be an integer or None (wrf.ALL_TIMES).  I going to assume you're actually looping across all of the time indexes.  Support for iterable timeidx values will be included in a future release. 

Thanks.

**Comment by yingkaisha** [comment]
Hi bladwig,

The file has been uploaded to the incoming. For the timeidx, sorry I do simplify my code for reporting the issue. 

Not sure how the whole system works, but will the fix be added in the anaconda file channel ?

Thanks.

**Comment by bladwig1** [comment]
Hi yingkaisha,

Thank you for sending the file.  The problem with XTIME is what I suspected.  Your variables are referencing the coordinate variable XTIME, but XTIME isn't actually in the file.  I don't think WRF would do this by default, so either it was removed in post-processing, or your WRF system has been configured not to output this variable.  While I would strongly caution against remove coordinate variables in the future, I will see if I can put in a quick fix for it, since I have to fix the projection issue.  

I will let you know when it's available on conda-forge, hopefully within 24 hours.
